pose_format/pose.py

```diff
@@ -51,37 +51,27 @@ def normalize(self, info: PoseNormalizationInfo, scale_factor: float = 1):
     """
     Normalize the point to a fixed distance between two points
     """
-    mask = self.body.data.mask
     transposed = self.body.zero_filled().points_perspective()
 
     p1s = transposed[info.p1]
     p2s = transposed[info.p2]
 
-    if transposed.shape[1] == 0:
-      p1s = p1s[0]
-      p2s = p2s[0]
-    else:
-      p1s = ma.concatenate(p1s)
-      p2s = ma.concatenate(p2s)
-
     # Move all points so center is (0,0)
-    center = np.mean((p2s + p1s) / 2, axis=0)
+    center = ((p2s + p1s) / 2).mean(axis=(0, 1))
     self.body.data -= center
 
-    mean_distance = np.mean(distance_batch(p1s, p2s))
+    mean_distance = distance_batch(p1s, p2s).mean()
 
-    scale = scale_factor / mean_distance  # scale all points to dist/scale
+    scale = scale_factor / float(mean_distance)  # scale all points to dist/scale
 
     if round(scale, 5) != 1:
```

Comment:
afaik, this is because numpy has a weird way of handling numbers... it can be that after this operation, repeating it should be of `scale==1` , but in actuality, `scale=0.99999921323` or something like that.
I assume because there is over-precision that is not handled that well in numpy. `round(..., 5)` would make such a number equal 1. another way of asking this is: `if math.abs(1 - scale) < 1e-5`. Added a comment

Reply:
Ok, all good! Yes that is because of floating-point precision.

I mistakenly thought that `round(0.99999921323, 5)` would result in `0.99999`, but in fact the result is `1.0`. I think `math.isclose` or `np.isclose` would be better functions for this test.